src/js/modules/FormFeedback.js

```diff
@@ -0,0 +1,231 @@
+import React, { Component } from 'react';
+import { defineMessages, injectIntl } from 'react-intl';
+import { postFeedback } from 'js/helpers/fetchData';
+import { logEvent } from 'js/helpers/googleAnalytics';
+import SectionHeaderSerif from 'js/modules/SectionHeaderSerif';
+
+import emojiDisappointed from 'images/emojis/disappointed.png';
+import emojiSad from 'images/emojis/sad.png';
+import emojiNeutral from 'images/emojis/neutral.png';
+import emojiSlightlySmiling from 'images/emojis/slightly_smiling.png';
+import emojiGrinning from 'images/emojis/grinning.png';
+
+const i18nMessages = defineMessages({
+  step1title: {
+    id: 'FormFeedback.step1.title',
+    defaultMessage: 'How satisfied are you with this page?',
+  },
+  step2titlea: {
+    id: 'FormFeedback.step2.title.a',
+    defaultMessage: 'Thank you!',
+  },
+  step2titleb: {
+    id: 'FormFeedback.step2.title.b',
+    defaultMessage: 'How can we make this page better?',
+  },
+  step3title: {
+    id: 'FormFeedback.step3.title',
+    defaultMessage: 'Thank you for sharing feedback!',
+  },
+  step3button: {
+    id: 'FormFeedback.step3.button',
+    defaultMessage: 'Give more feedback',
+  },
+  emojiDisappointed: {
+    id: 'FormFeedback.emojiDisappointed.altText',
+    defaultMessage: 'Disappointed',
+  },
+  emojiSad: {
+    id: 'FormFeedback.emojiSad.altText',
+    defaultMessage: 'Sad',
+  },
+  emojiNeutral: {
+    id: 'FormFeedback.emojiNeutral.altText',
+    defaultMessage: 'Neutral',
+  },
+  emojiSlightlySmiling: {
+    id: 'FormFeedback.emojiSlightlySmiling.altText',
+    defaultMessage: 'Slightly Smiling',
+  },
+  emojiGrinning: {
+    id: 'FormFeedback.emojiGrinning.altText',
+    defaultMessage: 'Grinning',
+  },
+});
+
+
+class FormFeedback extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      step: 1,
+      loading: false,
+      emoji: null,
+      feedback: null,
+      error: null
+    };
+  }
+
+  logEvent(action, label) {
+    logEvent({
+     category: 'FORM__site-feedback',
+     action: `${action}__step-${this.state.step}`,
+     label: label
+    });
+  }
+
+  handleEmojiClick = (e) => {
+    this.logEvent('emoji-click', JSON.stringify({
```

Comment:
does GA expect hyphens not camel case? I guess that doesn't matter just wondering.

Reply:
So would it be `emoji__click`? Why BEM if it doesn't relate to CSS?

Reply:
emoji-click cause BEM hyphenates compounds/phrases. Honestly, it can be anything. But with my GA experience, any naming convention is necessary for simplifying analysis later. Some of the biggest issues I've seen making GA data usable is that 20 different people had 20 different ideas as to how to name things, so aggregation was impossible. I chose BEM because we're already using it mostly, and because we're trying to name things related to html elements.